### ISSUE TITLE
SSL issues with `download.file()` and other libcurl-based request on R 4.2+

### ISSUE THREAD

**Issue by brownag** [ncss-tech/ncss-tech.github.io#3]
Since we switched over to R 4.2.1 many users have had issues with SSL connections preventing downloads and requests using the libcurl library which is used internally by base R. Examples of this include installing packages, downloading files inside soilDB functions, sourcing remote files, etc. This could have significant effects on getting sample datasets used in the stats for soil survey class.

As of 9/7/22 the _patched_ version of R 4.2.1 supports `R_LIBCURL_SSL_REVOKE_BEST_EFFORT` environment variable. 

### R-devel/NEWS for Wed, 07 Sep 2022
 > On Windows, environment variable ‘R_LIBCURL_SSL_REVOKE_BEST_EFFORT’ can be used to switch to only ‘best-effort’ SSL certificate revocation checks with the default ‘"libcurl"’ download method. This reduces security, but may be needed for downloads to work with MITM proxies (PR#18379).

In the future (if we get the patched version approved, or when we move to R 4.3.x) should consider adding to the default setup for CCE machines setting a user environment variable `R_LIBCURL_SSL_REVOKE_BEST_EFFORT=TRUE` to avoid people having to debug this themselves. While we have worked around it temporarily by using the R {curl} library and custom headers, there are many instances where it is impractical to use non-base R download facilities

**Comment by brownag** [comment]
I am pretty sure the common issues we encounter with SSL certificates (which may or may not be related to this issue that popped up a few years ago) are due to CRAN setting the `CURL_CA_BUNDLE` environment variable to a custom one associated with the R installation on Windows

This environment variable being set causes libcurl (used by e.g. PROJ, `download.file()`, and friends) to use the R bundled certificate file (i.e. `"C:\Program Files\R\R-4.4.2\etc\curl-ca-bundle.crt"`) as the root store and _**ignore**_ the corporate Windows certificate store.

This can be fixed with `Sys.unsetenv("CA_CURL_BUNDLE")`. Possibly this should be loaded into the .Rprofile or .Renviron as part of the stats class setup. I kinda loved not having to mess around with .Rprofile files and all the issues that come with potentially different home directories... but this might eliminate a lot of pain for people and save us from having to build solutions into e.g. soilDB to workaround.

**Comment by dylanbeaudette** [comment]
Good catch! I've been wondering about that too. Asking folks to use a simple .Rprofile worked in the past, so it is worth a shot.

**Comment by brownag** [comment]
So, upon further investigation with some folks that were having issues, unsetting the environment variable does not seem to resolve all of the SSL related issues. So, I think I will hold off on making any changes to the suggested setup on CCE machines.
